src/ControlzEx/Theming/WindowBackdropManager.cs

```diff
@@ -58,25 +58,34 @@ public static bool UpdateBackdrop(Window target, WindowBackdropType windowBackdr
                 return true;
             }
 
-            if (target is { AllowsTransparency: true })
-            {
-                SetCurrentBackdropType(target, WindowBackdropType.None);
-                return false;
-            }
-
-            if (FeatureSupport.IsWindowBackdropSupported is false)
-            {
-                SetCurrentBackdropType(target, WindowBackdropType.None);
-                return false;
-            }
-
             if (PresentationSource.FromVisual(target) is HwndSource hwndSource)
             {
+                if (hwndSource.CompositionTarget is { } compositionTarget)
```

Comment:
There is no need to change the background if backdrop isn't available or the window allows transparency as both of those values will never change during runtime.

Reply:
Yeah, but in our showcase we can

Reply:
We can't change AllowsTransparency after a window was shown and we can't change if backdrop support is available or not as that depends on the OS version. ;-)